plugins/GSdx/GSOsdManager.cpp

```diff
@@ -0,0 +1,394 @@
+#include "GSOsdManager.h"
+
+void GSOsdManager::LoadFont() {
+	FT_Error error = FT_New_Face(m_library, theApp.GetConfigS("fontname").c_str(), 0, &m_face);
+	if (error == FT_Err_Unknown_File_Format) {
+		fprintf(stderr, "Failed to init the freetype face: unknwon format\n");
+		return;
+	} else if (error) {
+		fprintf(stderr, "Failed to init the freetype face\n");
+		return;
+	}
+
+	LoadSize();
+}
+
+void GSOsdManager::LoadSize() {
+	m_size = theApp.GetConfigI("fontsize");
+	FT_Error error = FT_Set_Pixel_Sizes(m_face, 0, m_size);;
+	if (error) {
+		fprintf(stderr, "Failed to init the face size\n");
+		return;
+	}
+
+	/* This is not exact, I'm sure there's some convoluted way to determine these
+	 * from FreeType but they don't make it easy. */
+	atlas_w = m_size * 96; // random guess
+	atlas_h = m_size; // another random guess
+}
+
+GSOsdManager::GSOsdManager() : atlas_h(0)
+                             , atlas_w(0)
+                             , m_max_width(0)
+                             , m_texture_dirty(true)
+{
+	m_log_enabled = theApp.GetConfigB("log_enabled");
+	m_monitor_enabled = theApp.GetConfigB("monitor_enabled");
+	m_indicator_enabled = theApp.GetConfigB("indicator_enabled");
+	m_osd_transparency = theApp.GetConfigI("osd_transparency");
+
+	if (FT_Init_FreeType(&m_library)) {
+		fprintf(stderr, "Failed to init the freetype library\n");
+		return;
+	}
+
+	LoadFont();
+}
+
+GSOsdManager::~GSOsdManager() {
+	FT_Done_FreeType(m_library);
+}
+
+GSVector2i GSOsdManager::get_texture_font_size() {
+	return GSVector2i(atlas_w, atlas_h);
+}
+
+void GSOsdManager::upload_texture_atlas(GSTexture* t) {
+	if(c_info.size() > 96) // we only reserved space for this many glyphs
+		fprintf(stderr, "More than 96 glyphs needed for OSD");
+
+	// This can be sped up a bit by only uploading new glyphs
+	int x = 0;
+	for(auto &pair : c_info) {
+		if(FT_Load_Char(m_face, pair.first, FT_LOAD_RENDER)) {
+			fprintf(stderr, "failed to load char '%c' aka %d\n", pair.first, pair.first);
+			continue;
+		}
+
+		// Size of char
+		pair.second.ax = m_face->glyph->advance.x >> 6;
+		pair.second.ay = m_face->glyph->advance.y >> 6;
+
+		pair.second.bw = m_face->glyph->bitmap.width;
+		pair.second.bh = m_face->glyph->bitmap.rows;
+
+		pair.second.bl = m_face->glyph->bitmap_left;
+		pair.second.bt = m_face->glyph->bitmap_top;
+
+		GSVector4i r(x, 0, x+pair.second.bw, pair.second.bh);
+		if (r.width())
+			t->Update(r, m_face->glyph->bitmap.buffer, m_face->glyph->bitmap.pitch);
+
+		if (r.width() > m_max_width) m_max_width = r.width();
+
+		pair.second.tx = (float)x / atlas_w;
+		pair.second.ty = (float)pair.second.bh / atlas_h;
+		pair.second.tw = (float)pair.second.bw / atlas_w;
+
+		x += pair.second.bw;
+	}
+
+	m_texture_dirty = false;
+}
+
+/* This is dumb in that it doesn't check for malformed UTF8. This function
+ * is not expected to operate on user input, but only on compiled in strings */
+void dumb_utf8_to_utf32(const char *utf8, char32_t *utf32, unsigned size) {
```

Comment:
I need 4.9 support for myself :p (Debian Jessie). However if possible, I would like
```
#if gcc_vec < 5
reimplement_the_std
#else
use_the_std
#endif
```
This way, in the future code could be removed otherwise it would stay forever.